lib/util/unicode/is-combining-character.js

```diff
@@ -0,0 +1,287 @@
+// THIS FILE WAS GENERATED BY 'tools/update-unicode-utils.js'
+"use strict";
+
+/**
+ * Check whether a given character is a combining mark or not.
+ * @param {number} c The character code to check.
+ * @returns {boolean} `true` if the character belongs to the category, one of `Mc`, `Me`, and `Mn`.
+ */
+module.exports = function isCombiningCharacter(c) {
+    return (
+        (c >= 0x300 && c <= 0x36f) ||
```

Comment:
the checking seems time-consuming, can we create a lookup table here? (it takes a little more memory, but can be reused.)